File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if(root == null){return List.of();}
        List<Integer> list = new LinkedList<>();
        BFS(root, list);
        return list;
    }
    
    private void BFS(TreeNode root, List<Integer> list){
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while(queue.size() > 0){
            TreeNode right = null;
            var l = queue.size();
            for(int i = 0; i < l; i++){
                var node = queue.poll();
                if(node != null){
                    right = node;
                    queue.add(node.left);
                    queue.add(node.right);
                }
            }
            if(right != null){list.add(right.val);}
        }
    }
}

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if(root == NULL){
            return {};
        }
        vector<int> ans(0);
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            TreeNode* node = NULL;
            TreeNode* last = NULL;
            int len = q.size();
            for(int i = 0; i < len; i++){
                node = q.front();
                q.pop();
                if(node != NULL){
                    q.push(node->left);
                    q.push(node->right);
                    last = node;
                }
            }
            if(last != NULL){
                ans.push_back(last->val);
            }
        }
        return ans;
    }
};